js_modules/dagit/packages/core/src/runs/RunTimeline.tsx

```diff
@@ -0,0 +1,323 @@
+import * as React from 'react';
+import {Link} from 'react-router-dom';
+import styled from 'styled-components/macro';
+
+import {TimezoneContext} from '../app/time/TimezoneContext';
+import {browserTimezone} from '../app/time/browserTimezone';
+import {RunStatus} from '../types/globalTypes';
+import {ColorsWIP} from '../ui/Colors';
+import {FontFamily} from '../ui/styles';
+
+const ROW_HEIGHT = 24;
+const TIME_HEADER_HEIGHT = 36;
+const LABEL_WIDTH = 232;
+
+const ONE_HOUR_MSEC = 60 * 60 * 1000;
+
+type RunForJob = {
+  id: string;
+  status: RunStatus;
+  startTime: number;
+  endTime: number;
+};
+
+interface Props {
+  jobs: {[jobKey: string]: RunForJob[]};
+  range: [number, number];
+}
+
+export const RunTimeline = (props: Props) => {
+  const {jobs, range} = props;
+
+  const jobList = React.useMemo(() => {
+    const jobKeys = Object.keys(jobs);
+    const earliest = jobKeys.reduce((accum, jobKey) => {
+      const startTimes = jobs[jobKey].map((job) => job.startTime || 0);
+      return {...accum, [jobKey]: Math.min(...startTimes)};
+    }, {} as {[jobName: string]: number});
+
+    return jobKeys
+      .sort((a, b) => earliest[a] - earliest[b])
+      .map((jobKey) => ({jobKey, runs: jobs[jobKey]}));
+  }, [jobs]);
+
+  const height = ROW_HEIGHT * jobList.length;
+
+  return (
+    <Timeline $height={height}>
```

Comment:
This looks awesome! If we end up needing this to scroll horizontally I think we could drop an overflow:auto on this and use `const {viewport, containerProps} = useViewport();` to scope rendering to the visible jobs + runs the same way we do on the Gantt chart. Might not be necessary though if this always shows 24h 🙏